doc/user/content/transform-data/strategies-considerations.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Strategies and considerations"
+description: ""
+menu:
+  main:
+    parent: "transform-data"
+    weight: 95
+    name: "Strategies and considerations"
+---
+
+Materialize provides various [indexing strategies to optimize query
+performance](/transform-data/optimization/). However, because indexes in
+Materialize maintain the complete query results in memory (not just the index
+keys and the pointers to data rows), ... [will be updated].
+
+The following lists some strategies and considerations that may help you better
+utilize the benefits provided by Materialize while managing
+[costs](/administration/billing/).
+
+## Strategies to reduce dataset/memory usage
+
+### Projections
+
+Indexes store the complete query results in memory, not just the index keys and
+the pointers to data rows. As such, avoid extraneous fields in your query
+projections. That is, if you are interested in a smaller
+subset of the fields in your dataset, select only those fields instead of using
+`SELECT *`.
+
+### Temporal filters
+
+If you are only interested in the data from a recent period of time (e.g., the
+last 7 days, the last 24 hours, etc.), you can use [temporal
+filters](/transform-data/patterns/temporal-filters) to create a [sliding
+window](/transform-data/patterns/temporal-filters/#sliding-window) over the
+dataset.
+
+Considerations:
+
+- Determine if you need to add a grace period in order to account for late
+  arriving data (e.g., records that arrive out of order due to network issues).
+
+- If using stacked views with temporal filters, avoid unnecessarily duplicating
```

Comment:
Seems worth double checking with engineering. I tried to think through it but confused myself, but I don't think this is true. My reasoning is as follows: let's say we have something like
```sql
CREATE MATERIALIZED VIEW b AS SELECT * FROM a WHERE mz_now <= ts + '1 hour';
CREATE MATERIALIZED VIEW c AS SELECT * FROM b WHERE mz_now <= ts + '1 hour';
```
Because of the temporal filter in `b`, there is an insert and retraction for all rows in `a`. Similarly, the same is true for results in `c` (and there cannot be more retractions than inserts). But this means that even if the temporal filter is removed from `c`, there must be a single retraction. So the additional filter does not seem to make things worse. But I could be missing something.

Reply:
> Similarly, the same is true for results in c (and there cannot be more retractions than inserts).

That's actually not correct. Retractions from `b` do not necessarily make their way into `c`, e.g. arrangements hold onto future retractions and do not forward them. When two identical temporal filters are used, there is logic to prevent adding retractions multiple times. But when the temporal filters are different, we need to track future retractions in `b` and `c`.